bob/bob_test.rb

```diff
@@ -0,0 +1,116 @@
+    remark = 'Tom-ay-to, tom-aaaah-to.'
+    assert_equal 'Whatever.', bob.hey(remark), feedback(remark)
+  end
+
+  def test_shouting
+    remark = 'WATCH OUT!'
+    assert_equal 'Whoa, chill out!', bob.hey(remark), feedback(remark)
+  end
+
+  def test_shouting_gibberish
+    remark = ('A'..'Z').to_a.sample(10).join
+    assert_equal 'Whoa, chill out!', bob.hey(remark), feedback(remark)
+  end
+
+  def test_asking_a_question
+    remark = 'Does this cryogenic chamber make me look fat?'
+    assert_equal 'Sure.', bob.hey(remark), feedback(remark)
+  end
+
+  def test_asking_a_numeric_question
+    remark = 'You are, what, like 15?'
+    assert_equal 'Sure.', bob.hey(remark), feedback(remark)
+  end
+
+  def test_asking_gibberish
+    remark = ('a'..'z').to_a.sample(10).join << '?'
+    assert_equal 'Sure.', bob.hey(remark), feedback(remark)
+  end
+
+  def test_talking_forcefully
+    remark = "Let's go make out behind the gym!"
+    assert_equal 'Whatever.', bob.hey(remark), feedback(remark)
+  end
+
+  def test_using_acronyms_in_regular_speech
+    remark = "It's OK if you don't want to go to the DMV."
+    assert_equal 'Whatever.', bob.hey(remark), feedback(remark)
+  end
+
+  def test_forceful_questions
+    remark = 'WHAT THE HELL WERE YOU THINKING?'
+    assert_equal 'Whoa, chill out!', bob.hey(remark), feedback(remark)
+  end
+
+  def test_shouting_numbers
+    remark = '1, 2, 3 GO!'
+    assert_equal 'Whoa, chill out!', bob.hey(remark), feedback(remark)
+  end
+
+  def test_only_numbers
+    remark = '1, 2, 3'
+    assert_equal 'Whatever.', bob.hey(remark), feedback(remark)
+  end
+
+  def test_question_with_only_numbers
+    remark = '4?'
+    assert_equal 'Sure.', bob.hey(remark), feedback(remark)
+  end
+
+  def test_shouting_with_special_characters
+    remark = 'ZOMG THE %^*@#$(*^ ZOMBIES ARE COMING!!11!!1!'
+    assert_equal 'Whoa, chill out!', bob.hey(remark), feedback(remark)
+  end
+
+  def test_shouting_with_no_exclamation_mark
+    remark = 'I HATE YOU'
+    assert_equal 'Whoa, chill out!', bob.hey(remark), feedback(remark)
+  end
+
+  def test_statement_containing_question_mark
+    remark = 'Ending with ? means a question.'
+    assert_equal 'Whatever.', bob.hey(remark), feedback(remark)
+  end
+
+  def test_prattling_on
+    remark = 'Wait! Hang on. Are you going to be OK?'
+    assert_equal 'Sure.', bob.hey(remark), feedback(remark)
+  end
+
+  def test_silence
+    remark = ''
+    assert_equal 'Fine. Be that way!', bob.hey(remark), feedback(remark)
+  end
+
+  def test_prolonged_silence
+    remark = ' ' * rand(1..10)
+    assert_equal 'Fine. Be that way!', bob.hey(remark), feedback(remark)
+  end
+
+  def test_alternate_silences
+    remark = "\t" * rand(1..10)
+    assert_equal 'Fine. Be that way!', bob.hey(remark), feedback(remark)
+  end
+
+  def test_on_multiple_line_questions
+    remark = %(
+Does this cryogenic chamber make me look fat?
+no)
+    assert_equal 'Whatever.', bob.hey(remark), feedback(remark)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.